File: SwimmingCompetitionPersistance/src/main/java/ro/mpp/Validator/AngajatValidator.java
package ro.mpp.Validator;


import ro.mpp.Domain.Angajat;

public class AngajatValidator implements AbstractValidator<Angajat> {
    @Override
    public void validate(Angajat angajat) {
        String errors = "";

        if(angajat.getNume().isEmpty())
            errors += "Nume invalid!\n";
        if(angajat.getPrenume().isEmpty())
            errors += "Prenume invalid!\n";
        if(angajat.getParola().isEmpty())
            errors += "Parola invalida!\n";
        if(angajat.getLocatie().isEmpty())
            errors += "Locatie invalida!\n";
        if(!errors.isEmpty())
            throw new RuntimeException(errors);
    }
}

File: SwimmingCompetitionCSharpGUI/Validator/AngajatValidator.cs
using SwimmingCompetition.Domain;

namespace SwimmingCompetition.Validator;

public class AngajatValidator: IAbstractValidator<Angajat>
{
    public void Validate(Angajat e)
    {
        String errors = "";
        if (String.IsNullOrEmpty(e.Nume))
            errors += "Nume invalid!\n";
        if (String.IsNullOrEmpty(e.Prenume))
            errors += "Prenume invalid!\n";
        if (String.IsNullOrEmpty(e.Parola))
            errors += "Parola invalida!\n";
        if (String.IsNullOrEmpty(e.Locatie))
            errors += "Locatie invalida!\n";
        if(String.IsNullOrEmpty(errors))
            throw new Exception(errors);
    }
}